aea/helpers/dialogue/base.py

```diff
@@ -132,48 +134,153 @@ def __str__(self):
         )
 
 
-class Dialogue:
+class Dialogue(ABC):
     """The dialogue class maintains state of a dialogue and manages it."""
 
-    def __init__(self, dialogue_label: DialogueLabel) -> None:
+    STARTING_MESSAGE_ID = 1
+    STARTING_TARGET = 0
+
+    class Role(Enum):
+        """This class defines the agent's role in a dialogue."""
+
+        def __str__(self):
+            """Get the string representation."""
+            return self.value
+
+    class EndState(Enum):
+        """This class defines the end states of a dialogue."""
+
+        def __str__(self):
+            """Get the string representation."""
+            return self.value
+
+    def __init__(self, dialogue_label: DialogueLabel, role: Role) -> None:
         """
-        Initialize a dialogue label.
+        Initialize a dialogue.
 
         :param dialogue_label: the identifier of the dialogue
+        :param role: the role of the agent this dialogue is maintained for
 
         :return: None
         """
         self._dialogue_label = dialogue_label
+        self._role = role
+
         self._is_self_initiated = (
             dialogue_label.dialogue_opponent_addr
             is not dialogue_label.dialogue_starter_addr
         )
+
         self._outgoing_messages = []  # type: List[Message]
         self._incoming_messages = []  # type: List[Message]
 
     @property
     def dialogue_label(self) -> DialogueLabel:
-        """Get the dialogue lable."""
+        """
+        Get the dialogue label.
+
+        :return: The dialogue label
+        """
         return self._dialogue_label
 
+    @property
+    def role(self) -> "Role":
+        """
+        Get the agent's role in the dialogue.
+
+        :return: the agent's role
+        """
```

Comment:
`assert self._role is not None, "Role not set!"`

Reply:
Done